articles/cross_compilation_build_tools_design.md

```diff
@@ -0,0 +1,248 @@
+---
+layout: default
+title: Cross-compiling ROS 2 packages
+abstract:
+  This article is a design proposal for extending the build tools for ROS 2 to
+  simplify cross-compiling ROS 2 packages for platforms currently not supported
+  by ROS 2, or building on a host with different platforms than the target
+  platform. That later use case is particularly useful for speeding up build
+  times by building the code on a powerful development workstation, and then
+  deploying the binaries to a constrained resources hardware that uses the
+  target platform, thus making the development cycle much faster.
+author: >
+  [Thomas Moulard](https://github.com/thomas-moulard),
+  [Juan Rodriguez Hortala](https://github.com/juanrh)
+published: false
+---
+
+{:toc}
+
+# {{ page.title }}
+
+<div class="abstract" markdown="1">
+{{ page.abstract }}
+</div>
+
+Original Author: {{ page.author }}
+
+## Background
+
+The [cross-compilation tutorial](https://index.ros.org/doc/ros2/Tutorials/Cross-compilation)
```

Comment:
Just a minor note, both https://github.com/esteve/ros2_java and https://github.com/esteve/ros2_objc predate the instructions that Linaro wrote up. https://discourse.ros.org/t/crosscompiling-ros2-for-raspberry-pi-or-pretty-much-any-other-arm-based-board/2995 used the same technique as the Linaro instructions (Docker + QEmu), but for armhf as well.

Reply:
I'll add that precedents to the document, thanks for bringing them to our attention